backend/workers/lib/vod_converter/kitti.py

```diff
@@ -0,0 +1,178 @@
+"""
+Ingestor for KITTI formats.
+
+http://www.cvlibs.net/datasets/kitti/eval_object.php
+
+Per devkit docs:
+
+All values (numerical or strings) are separated via spaces,
+each row corresponds to one object. The 15 columns represent:
+
+#Values    Name      Description
+----------------------------------------------------------------------------
+   1    type         Describes the type of object: 'Car', 'Van', 'Truck',
+                     'Pedestrian', 'Person_sitting', 'Cyclist', 'Tram',
+                     'Misc' or 'DontCare'
+   1    truncated    Float from 0 (non-truncated) to 1 (truncated), where
+                     truncated refers to the object leaving image boundaries
+   1    occluded     Integer (0,1,2,3) indicating occlusion state:
+                     0 = fully visible, 1 = partly occluded
+                     2 = largely occluded, 3 = unknown
+   1    alpha        Observation angle of object, ranging [-pi..pi]
+   4    bbox         2D bounding box of object in the image (0-based index):
+                     contains left, top, right, bottom pixel coordinates
+   3    dimensions   3D object dimensions: height, width, length (in meters)
+   3    location     3D object location x,y,z in camera coordinates (in meters)
+   1    rotation_y   Rotation ry around Y-axis in camera coordinates [-pi..pi]
+   1    score        Only for results: Float, indicating confidence in
+                     detection, needed for p/r curves, higher is better.
+
+
+"""
+
+import csv
+import os
+import shutil
+
+from PIL import Image
+
+from .abstract import Ingestor, Egestor
+from .labels_and_aliases import output_labels
+
+
+class KITTIIngestor(Ingestor):
+    def validate(self, path, folder_names):
+        expected_dirs = [
+            'images',
+            'labels'
+        ]
+        for subdir in expected_dirs:
+            if not os.path.isdir(f"{path}/{subdir}"):
+                return False, f"Expected subdirectory {subdir} within {path}"
+        if not os.path.isfile(f"{path}/train.txt"):
+            return False, f"Expected train.txt file within {path}"
+        return True, None
+
+    def ingest(self, path, folder_names):
+        image_ids = self._get_image_ids(path)
+        image_ext = 'png'
+        if len(image_ids):
+            first_image_id = image_ids[0]
+            image_ext = self.find_image_ext(path, first_image_id)
+        tmp = [self._get_image_detection(path, image_name, image_ext=image_ext, folder_names=folder_names) for image_name in image_ids]
+        print('size: ' + str(len(tmp)))
```

Comment:
use logger

Reply:
sys.stdout was temporarily redirected to task.info to omit the need for replacing all prints in new scripts